### ISSUE TITLE
[question] bridging Meshroom and COLMAP?

### ISSUE THREAD

**Issue by devernay** [alicevision/meshroom#1395]
**Describe the problem**
It would be nice to be able to use alternative processings from COLMAP for *some* of the nodes in the Meshroom graph, so that one can switch easily between the two options for - say - SfM, or densification. I like the flexibility of the Meshroom graph, and I think it would be great to have COLMAP nodes in it.

Did someone from the community take a look at that already?

It doesn't seem like a huge amount of work, except maybe the translation between AliceVision and COLMAP SfM representations (and probably someone already has scripts to convert the camera poses between the two software - pointers?)

**Comment by natowi** [comment]
Yes, I played around with different packages, but at the moment it is pretty much a one way street.
https://github.com/natowi/meshroom_external_plugins/wiki/MR-external-photogrammetry

Creating nodes for existing Colmap executables should be easy, but what we need first is a converter node that supports multiple sfm formats from different software packages. (in this case primarily sfm <> nvm conversion)

**Comment by devernay** [comment]
Great starting points, thanks!

The blender addon also has code to more-or-less convert between these formats https://github.com/stuarta0/blender-photogrammetry
Your package https://github.com/natowi/meshroom_external_plugins also has pointers to plenty of additional resources!

**Comment by natowi** [comment]
The repository by stuarta0 is great - most of the code is already there and only needs to be wrapped up into a Meshroom node.
With this done, we can easily add support for Colmap and many other packages.

In addition to my personal repository, were I collect information I find useful, we now have a new API section in the documentation: https://meshroom-manual.readthedocs.io/en/latest/feature-documentation/core/nodes.html#api

**Comment by SBCV** [comment]
Just stumbled over [this conversion library](https://github.com/naver/kapture) - might be relevant for this task.

**Comment by natowi** [comment]
https://github.com/meshroomHub
Meshroom Research has some Colmap Support https://github.com/meshroomHub/MeshroomResearch
The new Plugin system will allow for easier extensibility.
